plugins/octant/cmd/traceflow-plugin/main.go

```diff
@@ -0,0 +1,409 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/vmware-tanzu/octant/pkg/navigation"
+	"github.com/vmware-tanzu/octant/pkg/plugin"
+	"github.com/vmware-tanzu/octant/pkg/plugin/service"
+	"github.com/vmware-tanzu/octant/pkg/view/component"
+	"github.com/vmware-tanzu/octant/pkg/view/flexlayout"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+
+	opsv1alpha1 "github.com/vmware-tanzu/antrea/pkg/apis/ops/v1alpha1"
+	clientset "github.com/vmware-tanzu/antrea/pkg/client/clientset/versioned"
+	"github.com/vmware-tanzu/antrea/pkg/graphviz"
+)
+
+var (
+	pluginName                           = "traceflowPlugin"
+	addTfAction                          = "traceflowPlugin/addTf"
+	showGraphAction                      = "traceflowPlugin/showGraphAction"
+	client          *clientset.Clientset = nil
+	kubeConfig                           = "KUBECONFIG"
+)
+
+const (
+	tfNameCol       = "Trace"
+	srcNamespaceCol = "Source Namespace"
+	srcPodCol       = "Source Pod"
+	srcPortCol      = "Source Port"
+	dstNamespaceCol = "Destination Namespace"
+	dstPodCol       = "Destination Pod"
+	dstSvcCol       = "Destination Service"
+	dstPortCol      = "Destination Port"
+	protocolCol     = "Protocol"
+	crdCol          = "Detailed Information"
+	phaseCol        = "Phase"
+	ageCol          = "Age"
+
+	octantTraceflowCRDPath = "/cluster-overview/custom-resources/traceflows.ops.antrea.tanzu.vmware.com/v1alpha1/"
+
+	namespaceStrPattern = `[a-z0-9]([-a-z0-9]*[a-z0-9])?`
+	podStrPattern       = `[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*`
+
+	TIME_FORMAT_YYYYMMDD_HHMMSS = "20060102-150405"
+)
+
+// According to code in Antrea agent and controller, default protocol is ICMP if protocol is not inputted by users.
+const (
+	ICMPProtocol int32 = 1
+	TCPProtocol  int32 = 6
+	UDPProtocol  int32 = 17
+)
+
+var supportedProtocols = map[string]int32{
+	"ICMP": ICMPProtocol,
+	"TCP":  TCPProtocol,
+	"UDP":  UDPProtocol,
+}
+
+// This is octant-trace-plugin.
+// The plugin needs to run with octant version v0.10.2 or later.
+func main() {
+	localPlugin := newTraceflowPlugin()
+
+	// Remove the prefix from the go logger since Octant will print logs with timestamps.
+	log.SetPrefix("")
+
+	capabilities := &plugin.Capabilities{
+		ActionNames: []string{addTfAction, showGraphAction},
+		IsModule:    true,
+	}
+
+	// Set up what should happen when Octant calls this plugin.
+	options := []service.PluginOption{
+		service.WithActionHandler(localPlugin.actionHandler),
+		service.WithNavigation(localPlugin.navHandler, localPlugin.initRoutes),
+	}
+
+	p, err := service.Register(pluginName, "A plugin that helps trace the packet flows and draw graphs for flows.", capabilities, options...)
```

Comment:
A plugin that starts Antrea Traceflow sessions to trace packets in the Antrea network and draws graphs for the result packet flows.

Reply:
Description changed.